### PR TITLE
[luci/logex] Replace SummaryBuilderLet::ABC

### PR DESCRIPTION
**Pull request by llFreetimell**
This commit replaces `SummaryBuilderLet::ABC` with `CircleNodeSummaryBuilders`.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

---

Parent Issue : #8357 
Draft : #8358

### CODE REVIEW

File: compiler/luci/logex/src/CircleNodeSummaryBuilders.cpp [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+namespace luci
+{
+
+std::vector<std::string> CircleNodeWithXSummaryBuilder::get_input_names(const luci::CircleNode *)
+{
+  return {"x"};
+}
+
+std::vector<std::string>
+CircleNodeWithINPUTSummaryBuilder::get_input_names(const luci::CircleNode *)
```
Comment (mhs4670go) [inline compiler/luci/logex/src/CircleNodeSummaryBuilders.cpp:123]: Is it intentional?
```suggestion
CircleNodeWithInputSummaryBuilder::get_input_names(const luci::CircleNode *)
```

**Review by mhs4670go — APPROVED** [review]
LGTM

File: compiler/luci/logex/src/CircleNodeSummaryBuilders.cpp (hunk 1 above)
Comment (llFreetimell) [reply compiler/luci/logex/src/CircleNodeSummaryBuilders.cpp:123]: I wrote it as upper case to represent `input` of `node->input()`.
But if it looks weird, I can apply your suggestion :D

Comment (mhs4670go) [reply compiler/luci/logex/src/CircleNodeSummaryBuilders.cpp:123]: Both way looks okay to me:)

Comment (llFreetimell) [reply compiler/luci/logex/src/CircleNodeSummaryBuilders.cpp:123]: Then I will go as it is :D

**Pull request merged**